code/datums/components/throwable_structure.dm

```diff
@@ -0,0 +1,182 @@
+			carbon_target.take_bodypart_damage(throw_force, 0, check_armor = TRUE, wound_bonus = 5)
+		else
+			living_target.apply_damage(throw_force, BRUTE, blocked = blocked, forced = FALSE, attack_direction = get_dir(get_turf(parent), living_target))
+		living_target.Knockdown(1.5 SECONDS)
+		playsound(parent, impact_sound, 40)
+	else
+		playsound(hit_atom, 'sound/weapons/genhit.ogg',80 , TRUE, -1)
+	return
+
+/datum/component/throwable_structure/proc/on_mouse_drop(datum/source, atom/dropping, mob/user)
+	SIGNAL_HANDLER
+	var/obj/structure_interact = parent
+	if(dropping == user && structure_interact.Adjacent(user))
+		if(structure_interact.anchored || structure_interact.flags_1 & NODECONSTRUCT_1)
+			return
+		if(!user.can_perform_action(structure_interact, NEED_DEXTERITY|NEED_HANDS))
+			return
+		INVOKE_ASYNC(src, PROC_REF(structure_try_pickup), user)
+
+/datum/component/throwable_structure/proc/structure_try_pickup(mob/living/user)
+	var/obj/structure_interact = parent
+	if(!ishuman(user))
+		return
+	if(user.get_num_held_items())
+		to_chat(user, span_warning("Your hands are full!"))
+		return FALSE
+	user.visible_message(span_notice("[user] starts picking up [parent]!"), span_notice("You start picking up [parent]..."))
+	if(!do_after(user, 3 SECONDS, target = parent))
+		return FALSE
+	if(structure_interact.anchored || !structure_interact.Adjacent(user) || user.get_num_held_items())
+		return FALSE
+	structure_pickup(user)
+	return TRUE
+
+/datum/component/throwable_structure/proc/structure_pickup(mob/living/user)
+	var/obj/item/structure_holder/holder = new(get_turf(parent), parent, held_state, held_lh, held_rh, held_force, throw_force, held_slowdown)
+	user.visible_message(span_warning("[user] picks up [parent]!"), span_warning("You pick up [parent]!"))
+	user.put_in_hands(holder)
+	playsound(holder, SFX_RUSTLE, 50, TRUE, -5)
+
+// Holder so we can pick it up
+/obj/item/structure_holder
+	name = "bugged structure holder"
+	desc = "Yell at coders."
+	icon = null
+	icon_state = ""
+	w_class = WEIGHT_CLASS_HUGE
+	attack_verb_continuous = list("smacks", "strikes", "cracks", "beats")
+	attack_verb_simple = list("smack", "strike", "crack", "beat")
+	slowdown = 1
+	item_flags = SLOWS_WHILE_IN_HAND
+	var/obj/held_structure
+	var/destroying = FALSE
+	var/datum/component/two_handed/two_handed_component
+
+/obj/item/structure_holder/Initialize(mapload, obj/target_structure, held_state, lh_icon, rh_icon, held_force, held_throwforce, held_slowdown)
+	if(held_state)
+		inhand_icon_state = held_state
+	if(lh_icon)
+		lefthand_file = lh_icon
+	if(rh_icon)
+		righthand_file = rh_icon
+	if(held_force)
+		force = held_force
+	if(held_throwforce)
+		throwforce = held_throwforce
+	if(held_slowdown)
+		slowdown = held_slowdown
+	deposit(target_structure)
+	two_handed_component = AddComponent(/datum/component/two_handed, require_twohands = TRUE, force_unwielded = force, force_wielded = force)
+	. = ..()
+
+/obj/item/structure_holder/Destroy()
+	destroying = TRUE
```

Comment:
the `destroying` var is unneeded and can be replaced with `QDELING(src)` instead